WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/data/WatchFaceUserSettingChangesDataSource.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.alpha.data
+
+import android.util.Log
+import androidx.wear.watchface.style.CurrentUserStyleRepository
+import androidx.wear.watchface.style.UserStyle
+import androidx.wear.watchface.style.UserStyleSetting
+import com.example.android.wearable.alpha.data.watchface.DRAW_HOUR_PIPS_DEFAULT
+import com.example.android.wearable.alpha.data.watchface.MINUTE_HAND_LENGTH_FRACTION
+import com.example.android.wearable.alpha.data.watchface.RED_COLOR_STYLE_ID
+import com.example.android.wearable.alpha.utils.COLOR_STYLE_SETTING
+import com.example.android.wearable.alpha.utils.DRAW_HOUR_PIPS_STYLE_SETTING
+import com.example.android.wearable.alpha.utils.WATCH_HAND_LENGTH_STYLE_SETTING
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+
+/**
+ * Data Source for all changes by the user in settings related to the watch face styles. Coverts
+ * the watch face api callback to a flow.
+ */
+class WatchFaceUserSettingChangesDataSource(
+    private val currentUserStyleRepository: CurrentUserStyleRepository
+) {
+    @ExperimentalCoroutinesApi
+    fun getUserWatchFaceChanges(): Flow<WatchFaceUserChanges> = callbackFlow {
+        val userStyleChangeListener = object : CurrentUserStyleRepository.UserStyleChangeListener {
```

Comment:
I think this could be a bit clearer if the conversion from a callback to a `Flow` happens as soon as possible. Or in other words, first constructing a `Flow<UserStyle>` directly from the callback, and then using `map` or other operators to transform the `Flow`.

That'd keep the `callbackFlow` section short.

Wrapping that into a helper of
```kotlin
val CurrentUserStyleRepository.userStyleChanges: Flow<UserStyle>
```
might be a good `ktx` candidate.

Reply:
This looks good to me!